zerver/tests/test_events.py

```diff
@@ -996,3 +997,15 @@ def test_use_narrow_if_default_is_none(self):
         self.user_profile.save()
         result = _default_narrow(self.user_profile, [])
         self.assertEqual(result, [])
+
+class AndroidDevGetEmailTest(AuthedTestCase):
+    def test_success(self):
+        # type: () -> None
+        result = self.client.get("/api/v1/dev_android_get_email")
+        self.assert_json_success(result)
+
+    def test_dev_auth_disabled(self):
+        # type: () -> None
+        with mock.patch('zerver.views.dev_auth_enabled', return_value=False):
+            result = self.client.get("/api/v1/dev_android_get_email")
+            self.assert_json_error_contains(result, "Dev environment not enabled.", 400)
```

Comment:
this test class should probably actually go in test_auth_backends next to the other tests.